cli_tools_tests/e2e/gce_ovf_export/test_suites/instance_ovf_export/instance_ovf_export_test_suite.go

```diff
@@ -0,0 +1,386 @@
+	err = computeClient.CreateInstanceBeta(testProjectConfig.TestProjectID, props.zone, &computeBeta.Instance{
+		Name:               props.instanceName,
+		SourceMachineImage: props.sourceGMI,
+	})
+
+	if err != nil {
+		e2e.Failure(testCase, logger, fmt.Sprintf("Error creating instance to export: %v", err))
+		return
+	}
+
+	instance, err := gcp.CreateInstanceObject(ctx, testProjectConfig.TestProjectID, props.zone, props.instanceName, props.isWindows)
+	if err != nil {
+		e2e.Failure(testCase, logger, fmt.Sprintf("Instance to export'%v' doesn't exist: %v", props.instanceName, err))
+		return
+	}
+
+	defer func() {
+		logger.Printf("Deleting temporary virtual machine instance `%v`", props.instanceName)
+		if err := instance.Cleanup(); err != nil {
+			logger.Printf("Temporary virtual machine instance '%v' failed to clean up: %v", props.instanceName, err)
+		} else {
+			logger.Printf("Temporary virtual machine instance '%v' cleaned up.", props.instanceName)
+		}
+	}()
+
+	logger.Printf("Exporting a temporary virtual machine instance `%v` to OVF at: %v", props.instanceName, props.destinationURI)
+	if e2e.RunTestForTestType(cmds[testType], args, testType, logger, testCase) {
+		if verifyExportedInstanceIsUnmodifiedAfterExport(ctx, testCase, testProjectConfig, logger, props) {
+			if ok, filesToClean := verifyExportedOVFContent(ctx, testCase, logger, props); ok {
+				defer cleanupGCSFiles(filesToClean, logger)
+				verifyExportedOVFUsingOVFImport(ctx, testCase, testProjectConfig, logger, props, computeClient)
+			}
+		}
+	}
+}
+
+func verifyExportedInstanceIsUnmodifiedAfterExport(
+	ctx context.Context, testCase *junitxml.TestCase, testProjectConfig *testconfig.Project,
+	logger *log.Logger, props *instanceOvfExportTestProperties) bool {
+
+	logger.Printf("Verifying exported instance is unmodified after export...")
+	instance, err := gcp.CreateInstanceObject(ctx, testProjectConfig.TestProjectID, props.zone, props.instanceName, props.isWindows)
+	if err != nil {
+		e2e.Failure(testCase, logger, fmt.Sprintf("Instance '%v' doesn't exist after export: %v", props.instanceName, err))
+		return false
+	}
+
+	attachedDisks := instance.Instance.Disks
+	if attachedDisks == nil || len(attachedDisks) != 3 {
+		e2e.Failure(testCase, logger, "Exported instance should have 3 disks attached after export")
+		return false
+	}
+
+	if instance.Instance.Status != "RUNNING" {
+		e2e.Failure(testCase, logger,
+			fmt.Sprintf(
+				"Exported instance should be in `RUNNING` state, but the state is `%v`", instance.Instance.Status))
+		return false
+	}
+	logger.Printf("Exported instance verified to be in the original sate...")
+	return true
+}
+
+func verifyExportedOVFContent(ctx context.Context, testCase *junitxml.TestCase,
+	logger *log.Logger, props *instanceOvfExportTestProperties) (bool, []*gcp.File) {
+
+	splits := strings.Split(props.sourceGMI, "/")
+	sourceGMIName := splits[len(splits)-1]
+	sourceGMIName = "machine-image-import-s2mtj"
```

Comment:
I'm guessing the second `sourceGMIName` was from a test run and can be removed?

Reply:
This was hardcoded based on the GMI used (which changed due to it being deleted by cleanerupper). 

I've refactored this to use instance disk device names, since that is used by the exported to generate file names.